src/routers/xcm/Router.sol

```diff
@@ -51,6 +37,10 @@ struct XcmWeightInfo {
     uint256 feeAmount;
 }
 
+interface GatewayLike {
+    function handle(bytes memory message) external;
+}
+
 contract ConnectorXCMRouter {
     using TypedMemView for bytes;
```

Comment:
Likewise
```suggestion
```

Reply:
Don't think I fully understand why we need this function, but `message_length_scale_encoded()` relies on `TypedMemView` and `ConnectorMessages`. I can remove bytes29 I think though.

Reply:
Crystal clear! I appreciate the explanation!